presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -119,6 +119,29 @@ public static Slice parseLine(@SqlType(StandardTypes.VARCHAR) Slice input)
         return serialize(geometry);
     }
 
+    @Description("Returns a LineString from an array of points created with ST_Point")
+    @ScalarFunction("ST_LineString")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stLineString(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input) {
+        if (input.getPositionCount() == 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "LineString cannot have one point");
+        }
+
+        MultiPath multipath = new Polyline();
+        for (int i = 0; i < input.getPositionCount(); i++) {
+            Slice slice = input.getSlice(i, 0, input.getSliceLength(i));
+            validateType("ST_Point", deserialize(slice), EnumSet.of(POINT));
```

Comment:
First argument needs to be the function name, e.g. `ST_LineString`. Could you add a negative test to verify the error message?

Reply:
I was looking at the implementation of validateType and was not sure that it would make sense if I put in ST_LineString, since it was actually just validating each individual point.  I believe the message would be "ST_LineString only applies to POINT, input type is POINT"".  The real issue is that the array should only include POINTS.

Reply:
If I check against the specific error message text, wouldn't we have to update a bunch of tests if we ever wanted to update the error message in validateType()?

Reply:
@arbass22 Yes, we would. It is better to do that, then not test the error message and not know that it doesn't make sense until a user files an issue.

Reply:
Alright, I will update the tests to check the error message